test/gtest/unity/util/GioMemory/GioMemory_test.cpp

```diff
@@ -64,9 +64,15 @@ class GioMemoryTest: public testing::Test
     {
         auto address = unique_glib(g_dbus_address_get_for_bus_sync(G_BUS_TYPE_SESSION, nullptr, nullptr));
 
+        GError *error = NULL;
         auto bus = unique_gobject(
                 g_dbus_connection_new_for_address_sync(address.get(), (GDBusConnectionFlags) (G_DBUS_CONNECTION_FLAGS_AUTHENTICATION_CLIENT | G_DBUS_CONNECTION_FLAGS_MESSAGE_BUS_CONNECTION), nullptr,
-                        nullptr, nullptr));
+                        nullptr, &error));
+        if (bus.get() == NULL) {
```

Comment:
As this is C++11 code, `std::unique_ptr` has a `bool` operator, so you don't need to do the `.get()` to see if the container is empty. Also, I think we should use `nullptr` in the instantiation of `error` above (as for example it is being passed instead of `NULL` to the above called method), for consistency and this being C++11.

Reply:
fixed. I've never used C++ smart pointers before so this is all new to me :)